packages/vite/src/node/server/sourcemap.ts

```diff
@@ -61,7 +61,7 @@ export async function injectSourcesContent(
 
 export function genSourceMapUrl(map: SourceMap | string | undefined): string {
   if (typeof map !== 'string') {
-    map = JSON.stringify(map)
+    map = map ? JSON.stringify(map) : 'undefined'
   }
   return `data:application/json;base64,${Buffer.from(map).toString('base64')}`
 }
```

Comment:
It looks like this function is used to generate `# sourceMappingUrl=` and  `map.toUrl`. For the former, I think it might make sense to not `# sourceMappingUrl=` at all if it's undefined? The [spec](https://docs.google.com/document/d/1U1RGAehQwRypUTovF1KRlpiOFze0b-_2gc6fAH0KY0k/edit#heading=h.lmz475t4mvbx) isn't very clear about this, but it might make sense to not use it if there's no url in the first place 🤔 

For the latter, the code is always non-null so it should be fine. In that case, we could tighten up the types of `genSourceMapUrl(map: SourceMap | string | undefined)` to `genSourceMapUrl(map: SourceMap | string)`

Reply:
This function is currently called without receiving undefined, so I think the type here should be shrunk, what do you think?

Reply:
Yeah I was also suggesting that we should shrink the type.